workers/unity/Packages/io.improbable.gdk.core/.codegen/Source/Generators/Core/ComponentDiffStorageGenerator.cs

```diff
@@ -0,0 +1,236 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Improbable.Gdk.CodeGeneration.CodeWriter;
+using Improbable.Gdk.CodeGeneration.Model.Details;
+using NLog;
+
+namespace Improbable.Gdk.CodeGenerator
+{
+    public static class ComponentDiffStorageGenerator
+    {
+        private static readonly Logger Logger = LogManager.GetCurrentClassLogger();
+
+        public static string Generate(UnityComponentDetails componentDetails, string qualifiedNamespace)
+        {
+            var eventDetailsList = componentDetails.EventDetails;
+
+            return CodeWriter.Populate(cgw =>
+            {
+                cgw.UsingDirectives(
+                    "System.Collections.Generic",
+                    "System",
+                    "Improbable.Gdk.Core",
+                    "Improbable.Worker.CInterop"
+                );
+
+                cgw.Namespace(qualifiedNamespace, ns =>
+                {
+                    ns.Type($"public partial class {componentDetails.ComponentName}", partial =>
+                    {
+                        Logger.Trace($"Generating {qualifiedNamespace}.{componentDetails.ComponentName}.DiffComponentStorage class.");
+
+                        var classDefinition = new StringBuilder("public class DiffComponentStorage : IDiffUpdateStorage<Update>, IDiffComponentAddedStorage<Update>, IDiffAuthorityStorage");
+
+                        foreach (var ev in eventDetailsList)
+                        {
+                            var eventType = $"{ev.EventName}.Event";
+                            classDefinition.Append($"{Environment.NewLine}    , IDiffEventStorage<{eventType}>");
```

Comment:
This should be using the indenting helper

Reply:
disagree - as this is a lot clearer to read